hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -315,16 +319,17 @@ protected void writeInsertRecord(HoodieRecord<T> newRecord) throws IOException {
 
   protected void writeInsertRecord(HoodieRecord<T> newRecord, Schema schema, Properties prop)
       throws IOException {
-    if (writeRecord(newRecord, Option.of(newRecord), schema, prop, HoodieOperation.isDelete(newRecord.getOperation()))) {
+    if (writeRecord(newRecord, Option.empty(), Option.of(newRecord), schema, prop, HoodieOperation.isDelete(newRecord.getOperation()))) {
       insertRecordsWritten++;
     }
   }
 
   protected boolean writeRecord(HoodieRecord<T> newRecord, Option<HoodieRecord> combineRecord, Schema schema, Properties prop) throws IOException {
-    return writeRecord(newRecord, combineRecord, schema, prop, false);
+    return writeRecord(newRecord, Option.empty(), combineRecord, schema, prop, false);
   }
 
-  private boolean writeRecord(HoodieRecord<T> newRecord, Option<HoodieRecord> combineRecord, Schema schema, Properties prop, boolean isDelete) throws IOException {
+  private boolean writeRecord(HoodieRecord<T> newRecord, Option<HoodieRecord<T>> oldRecordOpt, Option<HoodieRecord> combineRecord, Schema schema, Properties prop,
```

Comment:
Can you add some javadoc or comment explaining the difference between all the args? Specifically, what is a combineRecord here?

Reply:
Addressed